src/components/fidelity_bond/LockdateForm.tsx

```diff
@@ -0,0 +1,136 @@
+
+export const lockdateToTimestamp = (lockdate: Api.Lockdate): number => {
+  const split = lockdate.split('-')
+  return Date.UTC(parseInt(split[0], 10), parseInt(split[1], 10) - 1, 1)
+}
+
+// a maximum of years for a timelock to be accepted
+// this is useful in simple mode - when it should be prevented that users
+// lock up their coins for an awful amount of time by accident.
+// in "advanced" mode, this can be dropped or increased substantially
+export const DEFAULT_MAX_TIMELOCK_YEARS = 10
+
+type YearsRange = {
+  min: number
+  max: number
+}
+
+export const toYearsRange = (min: number, max: number): YearsRange => {
+  if (max <= min) {
+    throw new Error('Invalid values for range of years.')
+  }
+  return { min, max }
+}
+
+const initialLockdate = (now: Date, range: YearsRange): Api.Lockdate => {
+  const year = now.getUTCFullYear()
+  const month = now.getUTCMonth()
+  return dateToLockdate(new Date(Date.UTC(year + Math.max(range.min + 1, 1), month, 1)))
+}
+
+interface LockdateFormProps {
+  initialValue?: Api.Lockdate
+  onChange: (lockdate: Api.Lockdate) => void
+  yearsRange?: YearsRange
+  now?: Date
+}
+
+const LockdateForm = ({
+  onChange,
+  now = new Date(),
+  yearsRange = toYearsRange(0, DEFAULT_MAX_TIMELOCK_YEARS),
+  initialValue = initialLockdate(now, yearsRange),
+}: LockdateFormProps) => {
+  const { t } = useTranslation()
+
+  const currentYear = useMemo(() => now.getUTCFullYear(), [now])
+  const currentMonth = useMemo(() => now.getUTCMonth() + 1, [now]) // utc month ranges from [0, 11]
+
+  const initialDate = new Date(lockdateToTimestamp(initialValue))
+  const [lockdateYear, setLockdateYear] = useState(initialDate.getUTCFullYear())
+  const [lockdateMonth, setLockdateMonth] = useState(initialDate.getUTCMonth() + 1)
+
+  const minMonth = useCallback(() => {
+    if (lockdateYear > currentYear + yearsRange.min) {
+      return 1
+    }
+
+    // "minMonth" can be '13' - which means it never is valid and user must adapt 'year'.
+    return currentMonth + 1
+  }, [lockdateYear, currentYear, currentMonth, yearsRange])
+
+  const isLockdateYearValid = useMemo(
+    () => lockdateYear >= currentYear + yearsRange.min && lockdateYear <= currentYear + yearsRange.max,
+    [lockdateYear, currentYear, yearsRange]
+  )
+  const isLockdateMonthValid = useMemo(() => lockdateMonth >= minMonth(), [lockdateMonth, minMonth])
+
+  useEffect(() => {
+    if (!isLockdateYearValid || !isLockdateMonthValid) return
+
+    const date = new Date(Date.UTC(lockdateYear, lockdateMonth - 1, 1))
+    onChange(dateToLockdate(date))
+  }, [lockdateYear, lockdateMonth, isLockdateYearValid, isLockdateMonthValid, onChange])
+
+  return (
+    <rb.Row>
+      <rb.Col xs={6}>
+        <rb.Form.Group className="mb-4" controlId="locktimeYear">
+          <rb.Form.Label form="fidelity-bond-form">
+            <Trans i18nKey="fidelity_bond.form_create.label_locktime_year">Year</Trans>
+          </rb.Form.Label>
+          <rb.Form.Control
```

Comment:
Let's use dropdowns here. We can't use the default date picker since Firefox and Safari won't support that (see #221) but 10 years and 12 months are an ok enough range for a dropdown.